### DISCUSSION TITLE
I want to convert color1 to color2 of all pixels in image.

### DISCUSSION THREAD

**Discussion by bemoregt** [ImageMagick/ImageMagick#2371] [Help]
Hi, All.

I want to convert color1 to color2 of all pixels in image.

My image have grayscale  colorspace.

example: convert  color:red color:1 aaa.png  ?

Thanks in advance.

Best,
@bemoregt.

**Comment by fmw42** [comment]
`convert aaa.png -fuzz XX% -fill color2 -opaque color1 result.png`

where XX is between 0 and 100 and most likely near 0. It is a tolerance for changing any value near color1 to color2. The larger the XX the more different values for color1 will change to color 2. 

See https://imagemagick.org/Usage/color_basics/#fuzz

Colors can be defined in different ways. See https://imagemagick.org/script/color.php
